### DISCUSSION TITLE
Pipeline resource signature missing resource with static inlined buffer data

### DISCUSSION THREAD

**Discussion by RobertMtx** [DiligentGraphics/DiligentEngine#253] [Q&A] [answered]
I'm not sure if this should be posted as an issue, because I'm not entirely sure if I'm setting things up correctly.

I'm trying to use the following block of code in a shader:
```
cbuffer POISSON_DISKS
{
	float2 poissonDisk[16] = {
		float2( -0.94201624, -0.39906216 ),
		float2( 0.94558609, -0.76890725 ),
		float2( -0.094184101, -0.92938870 ),
		float2( 0.34495938, 0.29387760 ),
		float2( -0.91588581, 0.45771432 ),
		float2( -0.81544232, -0.87912464 ),
		float2( -0.38277543, 0.27676845 ),
		float2( 0.97484398, 0.75648379 ),
		float2( 0.44323325, -0.97511554 ),
		float2( 0.53742981, -0.47373420 ),
		float2( -0.26496911, -0.41893023 ),
		float2( 0.79197514, 0.19090188 ),
		float2( -0.24188840, 0.99706507 ),
		float2( -0.81409955, 0.91437590 ),
		float2( 0.19984126, 0.78641367 ),
		float2( 0.14383161, -0.14100790 )
	};
};
```

But Diligent is complaining that I'm making use of this resource without linking it:

> Diligent Engine: ERROR in Diligent::PipelineStateD3D12Impl::ValidateShaderResources::<lambda_090d5528af80d42242819c8b19c8c8f5>::operator ()() (PipelineStateD3D12Impl.cpp, 589): Shader 'CGS_Standard_i3v0p8192a2147483648' contains resource 'POISSON_DISKS' that is not present in any pipeline resource signature used to create pipeline state 'CGS_Standard_i3v0p8192a2147483648'.

Is it necessary to link an inlined resource like this? I borrowed the poisson disks from NVidia, so its entirely possible they intended the programmer to take some other step in order to use it that I am not aware of.

**Comment by RobertMtx** [comment]
When I add it to my resource description as a static constant buffer, it seems to be okay with it. Is this the correct way to use such a resource?

**Reply by TheMostDiligent** [reply, marked as answer]
Inline constant buffers don't work, you need to create a buffer on the host and bind it

**Reply by RobertMtx** [reply]
Yeah, I figured this out eventually, but wasn't sure why they didn't work. Thanks for letting me know.

**Reply by TheMostDiligent** [reply]
They don't work because I have no idea how they are supposed to work. Shader code is defined at compile time. Buffer data is defined at run time. Inline buffers somehow assume that loading shaders will also involve creating buffers with data.
